src/accelerate/utils/dataclasses.py

```diff
@@ -1810,6 +1811,31 @@ def set_mixed_precision(self, mixed_precision, buffer_autocast=False, override=F
             self.mixed_precision_policy = MixedPrecision(**self.mixed_precision_policy)
 
 
+@dataclass
+class TorchTensorParallelPlugin:
+    """
+    This plugin is used to enable tensor parallelism using PyTorch 2.0.
+    """
+
+    tp_size: int = field(
+        default=1,
+        metadata={
+            "help": "tensor parallel size will be used in the device mesh preparation"
+        },
+    )
+
+    # type has to be "torch.distributed.DeviceMesh"
+    torch_device_mesh: torch.distributed.DeviceMesh = field(
+        default=None
+    )
+
+    def __post_init__(self):
+        from torch.distributed.device_mesh import init_device_mesh
```

Comment:
Should we perform a check on the minimum PyTorch and transformers versions? Not sure if here is the best place or somewhere else, Zach?

Reply:
I see, good point. Still, torch could be checked, right?

Reply:
Added torch version check thanks